### DISCUSSION TITLE
how to keep showing a loading spinner when the page is revalidating after a server action is invoked

### DISCUSSION THREAD

**Discussion by aaronvg** [vercel/next.js#50721] [App Router]
Basically I do a startTransition and invoke a server action. The action completes, but then it takes a while for the page to actually refresh. I want to show a spinner (basically isPending should still be true) if the page is in the process of revalidating. How would I do this?

**Comment by titiloxx** [comment]
+1

**Comment by dustinahudson** [comment]
This is similar to [this](https://github.com/vercel/next.js/discussions/50769). Not specific to server actions but having a way to indicate something is happening when upstream dependencies are slow is a requirement for my project. I was about ready to switch to server actions but it sounds looks like it won't help.

**Comment by Zain-ul-din** [comment]
+1
